### ISSUE TITLE
Missing error when derived class does not call `super` in initializer

### ISSUE THREAD

**Issue by Chriscbr** [winglang/wing#4277]
### I tried this:

```js
class Foo {
  new(name: str) {}
}

class Bar extends Foo {
  new() {}
}
```

### This happened:

It compiles

### I expected this:

A type checking error, saying that `Bar`'s initializer needs a `super` call since it is a derived class

### Component

Compiler

### Wing Version

0.33.3

### Node.js Version

18.14.1

### Platform(s)

MacOS

### Community Notes

* Please vote by adding a 👍 reaction to the issue to help us prioritize.
* If you are interested to work on this issue, please leave a comment.

**Comment by KrishnaSindhur** [comment]
i can take look if its till open?

**Comment by staycoolcall911** [comment]
Thanks @KrishnaSindhur, I assigned the issue to you. Please let us know if you have any questions or need pointers

**Comment by staycoolcall911** [comment]
This now correctly yields the desired error message (Wing V0.74.30):
```sh
Failed to compile.

error: Missing super() call as first statement of Bar's constructor
  --> wing/main.w:6:9
  |
6 |   new() {}
  |         ^^
```

**Comment by staycoolcall911** [comment]
Fixed in #6351
